datafusion/expr/src/udf.rs

```diff
@@ -202,18 +203,27 @@ impl ScalarUDF {
         Arc::new(move |args| captured.invoke(args))
     }
 
-    /// This function specifies monotonicity behaviors for User defined scalar functions.
-    ///
-    /// See [`ScalarUDFImpl::monotonicity`] for more details.
-    pub fn monotonicity(&self) -> Result<Option<FuncMonotonicity>> {
-        self.inner.monotonicity()
-    }
-
     /// Get the circuits of inner implementation
     pub fn short_circuits(&self) -> bool {
         self.inner.short_circuits()
     }
 
+    pub fn evaluate_bounds(&self, input: &[&Interval]) -> Result<Interval> {
```

Comment:
If we want to add these new functions to the scalar function API I think they need doc comments explaining what they do and how to use them